xapian-core/include/xapian/cluster.h

```diff
@@ -28,14 +28,43 @@
 #endif
 
 #include <xapian/mset.h>
+#include <xapian/queryparser.h>
 #include <xapian/types.h>
 #include <xapian/visibility.h>
 
 #include <unordered_map>
+#include <unordered_set>
 #include <vector>
 
 namespace Xapian {
 
+/** Implementation of the stopper class that can be used to identify
+ *  stemmed as well as unstemmed stopwords
+ */
+class XAPIAN_VISIBILITY_DEFAULT StemStopper : public Xapian::Stopper {
+    std::unordered_set<std::string> stop_words;
+    Xapian::Stem stemmer;
+
+  public:
+    /// Default constructor
+    StemStopper();
+
+    /** Set the stemmer.
+     *
+     *  @param stem	The Xapian::Stem object to set.
+     */
+    void set_stemmer(const Xapian::Stem &stemmer);
```

Comment:
API consistency is good (and good to think about), but it's not the only consideration.

As I said, it's not useful to change the stemmer mid-use, so providing both seems to just be providing a method for which there's no sensible use.  The difference for the other classes with `set_stemmer()` is that it is useful to be able to change the stemmer - e.g. I might use the same `QueryParser` object to parse many queries, and have a control in the UI which selects the stemming language to use.

A more extreme example - arguably it would be more consistent if every Xapian API class had a `set_stemmer()` method, since several classes do.  But this greater consistency doesn't help the API user - instead of remembering which classes support `set_stemmer()`, they instead need to remember which classes support `set_stemmer()` in a useful way...